app/services/catalog/update_order_item.rb

```diff
@@ -51,5 +45,16 @@ def fetch_external_url
       Rails.logger.error("Could not find an external url on service instance (id: #{@service_instance_id}) attached to task_id: #{@payload["task_id"]}")
       raise "Could not find an external url on service instance (id: #{@service_instance_id}) attached to task_id: #{@payload["task_id"]}"
     end
+
+    def mark_item_finished(order_item)
```

Comment:
@eclarizio Shouldn't there also be a failure case. The old order_item had mark_finished and mark_failed.

Reply:
Good point, looking at https://github.com/ManageIQ/topological_inventory-openshift/blob/master/lib/topological_inventory/openshift/operations/processor.rb#L48, https://github.com/ManageIQ/topological_inventory-openshift/blob/master/lib/topological_inventory/openshift/operations/processor.rb#L74, and https://github.com/ManageIQ/topological_inventory-openshift/blob/master/lib/topological_inventory/openshift/operations/processor.rb#L78, it seems like we always return "completed" for the state, but the `status` differs between "ok" and "error".

I've updated the code on our side to handle the "ok" and "error" statuses instead of looking at the state.